### PR TITLE
Expire node, set last version on node

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

**Comment by jayrbolton** [comment]
Higher level comment: can you make these queries more bulk-ish? Eg. could they take a list of keys rather than one key? I just worry about bulk performance here.

File: relation_engine/database/test/batch_time_travel_test.py [hunk 1]
```diff
@@ -82,4 +82,167 @@ def test_save_vertex(arango_db):
...
+    ret = att.get_vertex('bar', 200)
+    assert ret == {'_key': '2', 'id': 'bar', 'created': 100, 'expired': 9000}
+
+    _check_no_fake_changes(arango_db)
+
+    _check_docs(arango_db, edges1, 'edges1')
+    _check_docs(arango_db, edges2, 'edges2')
+
+def test_set_node_expired_vert_and_edges_one_collection(arango_db):
+
```
Comment (jayrbolton) [inline relation_engine/database/test/batch_time_travel_test.py]: Can you add comments in each of these test with a plain english description of what this is testing?

File: relation_engine/database/test/batch_time_travel_test.py [hunk 2]
```diff
@@ -82,4 +82,167 @@ def test_save_vertex(arango_db):
...
+    _, edges2 = _setup_set_node_expired(arango_db)
+    att = ArangoBatchTimeTravellingDB(arango_db, default_vertex_collection='verts')
+
+    att.expire_vertex('1', 500, ['edges1'])
```
Comment (jayrbolton) [inline relation_engine/database/test/batch_time_travel_test.py]: FYI, stack tracing the collection that this is actually updating took me a little while to do. I would personally prefer to have the collection name always go in these args, but it's up to you

**Comment by jayrbolton** [comment]
What happens if you expire a vertex that does not exist, but has edges using its ID?

**Comment by jayrbolton** [comment]
@MrCreosote i have no more comments

File: relation_engine/database/test/batch_time_travel_test.py (hunk 2 above)
Comment (MrCreosote) [reply relation_engine/database/test/batch_time_travel_test.py]: I feel like the name is going to be used in almost every method, so having a default can really reduce the amount of arguments across all the methods.

**Comment by MrCreosote** [comment]
> Higher level comment: can you make these queries more bulk-ish?

Yep, but I'll do that as necessary. Simple & working first, optimize after you figure out what parts are slow.

> What happens if you expire a vertex that does not exist, but has edges using its ID?

If you try and get it by the id and load timestamp, you won't get anything so the expiration won't happen. If you somehow get the `_key` some other way and expire it, it'll reset all the expire times to the provided expire time.

**Comment by MrCreosote** [comment]
Not sure if it's even possible to do bulk updates... you can def do bulk inserts though

File: relation_engine/database/test/batch_time_travel_test.py (hunk 1 above)
Comment (MrCreosote) [reply relation_engine/database/test/batch_time_travel_test.py]: Done

**Comment by MrCreosote** [comment]
@jayrbolton all comments addressed

**Comment by jayrbolton** [comment]
By bulk update, I just mean an aql "update" statement inside an aql for-loop that applies to multiple docs.
We might want to make sure that edges expire for a key even if the vertex does not exist. Lots of dangling edges in the workspace shadow space, for example. 
I don't have any other comments, feel free to merge.

**Comment by MrCreosote** [comment]
> By bulk update, I just mean an aql "update" statement inside an aql for-loop that applies to multiple docs.

Yeah, for the updates to be bulk we've have to have an `in` type match that's a list of `_key`s. I assume there's some way to do that.

> We might want to make sure that edges expire for a key even if the vertex does not exist. Lots of dangling edges in the workspace shadow space, for example.

Right now my plan is that the algorithm will be responsible for ensuring there's no dangling edges, but if that doesn't work out I can add more safety to the DB wrapper code here.

**Pull request merged**